### DISCUSSION TITLE
Programming Artery from scratch?

### DISCUSSION THREAD

**Discussion by StormTrooper** [keirf/greaseweazle#448] [Q&A] [answered]
I'm busy building a v4.1 from scratch and have a question on programming the at32f403. 

I'm a bit confused over the documentation. Should I be able to use a ST-Link v2 to program the Artery. I'm guessing not as it's not a STM32 micro.  (I've tried using my ST-Link anyway and unable to program it)

What programmer should I be using to program these chips? The Artery site seems to be equally confusing. I found a few references on using a AT-Link programmer?

**Comment by keirf** [comment, marked as answer]
Just follow the USB-DFU method https://github.com/keirf/Greaseweazle/wiki/Firmware-Programming#programming-v4-devices-via-usb-dfu

It requires no programmer, just Artery software (Windows only), and a USB cable.

If you must program on a different OS, come back to me.

**Reply by StormTrooper** [reply]
Thanks. Wasn't aware it could be programmed directly with the USB cable. 

Other issue I had is the 5V on USB  was jumpered :) , and then I was trying this on a Win7 VM. Either Win7 no longer supported by Artery (my guess) or it doesn't like the VM.

Ran it on a physical W11 machine and it's working 100%
